tests/foreman/api/test_imagemode.py

```diff
@@ -524,3 +526,256 @@ def test_bootc_rex_errata_install(target_sat, bootc_host, function_repos_collect
+
+    # Add only persistent packages
+    package_data = [
+        {
+            'name': 'base-pkg-1',
+            'version': '1.0.0',
+            'release': '1.el10',
+            'arch': 'x86_64',
+            'persistence': 'persistent',
+        },
+        {
+            'name': 'base-pkg-2',
+            'version': '2.0.0',
+            'release': '1.el10',
+            'arch': 'x86_64',
+            'persistence': 'persistent',
+        },
+    ]
+
+    _create_transient_packages(target_sat, host, package_data)
+
+    # Call the API endpoint
+    packages_url = f'{target_sat.url}/api/v2/hosts/{host.id}/packages'
+    auth = (settings.server.admin_username, settings.server.admin_password)
+    response = requests.get(packages_url, auth=auth, verify=False)
+
+    assert response.status_code == 200
+    packages = response.json()['results']
+
+    # Verify all packages are persistent
+    assert len(packages) == 2
+    for pkg in packages:
+        assert 'persistence' in pkg, f"Package {pkg['name']} missing persistence field"
+        assert pkg['persistence'] == 'persistent', (
+            f"Expected all packages to be persistent, but {pkg['name']} has persistence='{pkg['persistence']}'"
+        )
+
+
+def test_positive_host_packages_persistence_api_empty_results(target_sat):
+    """Verify that package persistence API returns an empty result set for a host with no packages
+
+    :id: d4e5f6a7-b8c9-0123-def0-1234567890ab
+
+    :steps:
+        1. Create a host
+        2. Ensure host has no packages in katello_host_installed_packages
+        3. Call /api/v2/hosts/:id/packages API endpoint
+        4. Verify response is successful with empty results
+
+    :expectedresults:
+        1. Response status is 200
+        2. Results field is present and is an empty list
+        3. Total and subtotal are 0
+
+    :Verifies: https://issues.redhat.com/browse/SAT-36788
+    """
+    # Create a host
+    host = target_sat.api.Host().create()
+
+    # Ensure this host has no package persistence rows (it shouldn't by default, but let's be explicit)
+    target_sat.query_db(
+        f"DELETE FROM katello_host_installed_packages WHERE host_id = {host.id}",
+        output_format="raw",
+    )
+
+    # Call the packages API for a host with no installed packages
+    packages_url = f'{target_sat.url}/api/v2/hosts/{host.id}/packages'
+    auth = (settings.server.admin_username, settings.server.admin_password)
+    response = requests.get(packages_url, auth=auth, verify=False)
+
+    # Response should be successful even when there are no packages
+    assert response.status_code == 200
+
+    body = response.json()
+
+    # results key must be present and an empty list
+    assert "results" in body
+    assert isinstance(body["results"], list)
+    assert body["results"] == []
+
+    # total/subtotal should correctly represent the empty case
+    assert body.get("total") == 0
+    assert body.get("subtotal") == 0
```

Comment:
We could use the NG even here. An example from python console:
```
>>> from robottelo.hosts import Satellite
>>> target_sat = Satellite('my_satellite_box.redhat.com')
>>> host = target_sat.api.Host().create()
>>> host.packages()
{'total': 0, 'subtotal': 0, 'selectable': 0, 'page': 1, 'per_page': 20, 'error': None, 'search': None, 'sort': {'by': 'name', 'order': 'asc'}, 'results': []}
>>> host.packages()['results']
[]
```

```suggestion
    packages = host.packages()

    # results key must be present and an empty list
    assert 'results' in packages
    assert isinstance(packages['results'], list)
    assert len(packages['results']) == 0

    # total/subtotal should correctly represent the empty case
    assert body.get("total") == 0
    assert body.get("subtotal") == 0
```